HW1/P4/P4.py

```diff
@@ -0,0 +1,76 @@
+# Start the cluster
+import seaborn as sns
+sns.set_context('poster', font_scale=1.25)
+import findspark as fs
+fs.init()
+import pyspark as ps
+import multiprocessing as mp
+
+### Setup ###
+config = ps.SparkConf()
+# Assumes setup is in spark.conf
+config = config.setAppName('marvel_solver')
+sc = ps.SparkContext(conf=config)
+
+# Read in the data
+
+marvel_raw_data = sc.textFile('marvel_data.csv')
+
+def get_issue_character(raw_line):
+    dat = raw_line.split('"')
+    character_name = dat[1]
+    issue = dat[3]
+    return (issue, character_name)
+
+issue_character_rdd = marvel_raw_data.map(get_issue_character)
+# Put everyone in a group if they are in the same comic and get nodes from that!
+issue_groups_rdd = issue_character_rdd.groupByKey()
+
+def get_links_from_groups(x):
```

Comment:
This is quite a costly operation, which I think could be parallelized using only rdds